moveit_ros/planning_interface/moveit_cpp/src/moveit_cpp.cpp

```diff
@@ -108,14 +108,16 @@ MoveItCpp::~MoveItCpp()
   clearContents();
 }
 
-MoveItCpp& MoveItCpp::operator=(MoveItCpp&& other)
+MoveItCpp& MoveItCpp::operator=(MoveItCpp&& other) noexcept
 {
   if (this != &other)
   {
     this->node_handle_ = other.node_handle_;
-    this->tf_buffer_ = other.tf_buffer_;
-    this->robot_model_ = other.robot_model_;
-    this->planning_scene_monitor_ = other.planning_scene_monitor_;
+    this->tf_buffer_ = std::move(other.tf_buffer_);
+    this->tf_listener_ = std::move(other.tf_listener_);
+    this->robot_model_ = std::move(other.robot_model_);
+    this->planning_scene_monitor_ = std::move(other.planning_scene_monitor_);
+    // TODO: what about the other elements?
```

Comment:
@henningkayser, can you comment on this?
Why are the other members of MoveItCpp not handled here?

Reply:
You're right, the other components should be moved as well. Thanks for looking into this. They were probably added later and moving MoveItCpp was never used in practice, so it wasn't noticed

Reply:
I can create a fixup PR

Reply:
Is there any reason the move or copy constructor needs to explicitly defined?  It looks like the default move constructor behavior is what is intended.  Why not use the default move constructor by putting this in the header?

```c++
MoveItCpp(const MoveItCpp& other)            = default;
MoveItCpp(MoveItCpp&& other)                 = default;
MoveItCpp& operator=(const MoveItCpp& other) = default;
MoveItCpp& operator=(MoveItCpp&& other)      = default;
```